test/common/index.js

```diff
@@ -83,10 +83,41 @@ function parseTestFlags(filename = process.argv[1]) {
   if (source[flagEnd - 1] === '\r') {
     flagEnd--;
   }
-  return source
-    .substring(flagStart, flagEnd)
-    .split(/\s+/)
-    .filter(Boolean);
+  const str = source.substring(flagStart, flagEnd);
+  const args = [];
+  let i = 0;
+  let start = 0;
+  let inString = false;
+  let inArg = false;
+
+  function unwrapArg(arg) {
+    const shouldTrim = (arg[0] === '"' && arg[arg.length - 1] === '"') ||
+                        (arg[0] === '\'' && arg[arg.length - 1] === '\'');
+    return shouldTrim ? arg.slice(1, -1) : arg;
+  }
+  while (i < str.length) {
+    const c = str[i];
+    if (c === '=' && !inString) {
+      args.push(str.substring(start, i));
+      start = i + 1;
+    } else if (c === ' ' && !inString) {
+      if (inArg) {
+        args.push(unwrapArg(str.substring(start, i)));
+        inArg = false;
+      }
+      start = i + 1;
+    } else if (c === '"' || c === '\'') {
```

Comment:
Wouldn't this fail if a string starts with `"` and then you see a `'` which should be part of the string/arg?